src/v/datalake/location.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#pragma once
+
+#include "cloud_io/provider.h"
+#include "datalake/base_types.h"
+#include "iceberg/uri.h"
+#include "model/fundamental.h"
+
+#include <optional>
+
+namespace datalake {
+
+class location_provider {
```

Comment:
nit: should this be some path_converter or something? It doesn't look unique to locations, unless it will eventually learn about table locations?

Reply:
This is inspired by pyiceberg/java iceberg 
- https://github.com/apache/iceberg-python/blob/main/pyiceberg/table/locations.py
- https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/LocationProviders.java

Reply:
Yea I've seen these classes, but I always associated them with giving paths for a data file based on properties of the table. That's why I found this class name surprising because it's doing a pretty different function from the others.